py-polars/src/lazyframe/visitor/nodes.rs

```diff
@@ -257,29 +257,41 @@ pub struct Sink {
 
 pub(crate) fn into_py(py: Python<'_>, plan: &IR) -> PyResult<PyObject> {
     let result = match plan {
-        IR::PythonScan { options, predicate } => PythonScan {
-            options: (
-                options
-                    .scan_fn
-                    .as_ref()
-                    .map_or_else(|| py.None(), |s| s.0.clone()),
-                options
-                    .with_columns
-                    .as_ref()
-                    .map_or_else(|| py.None(), |cols| cols.to_object(py)),
-                options.pyarrow,
-                options
-                    .predicate
-                    .as_ref()
-                    .map_or_else(|| py.None(), |s| s.to_object(py)),
-                options
-                    .n_rows
-                    .map_or_else(|| py.None(), |s| s.to_object(py)),
-            )
-                .to_object(py),
-            predicate: predicate.as_ref().map(|e| e.into()),
-        }
-        .into_py(py),
+        IR::PythonScan { options } => {
+            let python_src = match options.python_source {
```

Comment:
@wence- technically I must increase the major version here, but I cannot imagine this is used on your side. Would like to leave it as is.

Reply:
We translate pythonscan [right now](https://github.com/rapidsai/cudf/blob/branch-24.10/python/cudf_polars/cudf_polars/dsl/translate.py#L75), but leave it unimplemented. But this would break the translation if we encounter a pythonscan (fallback should work fine though), rather than us raising a "useful" error, since there's no longer a `predicate` slot separate from the `options.predicate`.

If you bump the major version, we can then handle it in translation "transparently" on our side.

FWIW, I'd prefer a major version bump.